design_pim/overview.rst

```diff
@@ -0,0 +1,354 @@
+
+.. code-block:: javascript
+    :linenos:
+
+    {
+        logLocale: function () {
+            console.log(UserContext.get('uiLocale'));
+        }
+    }
+
+
+Our newly created object has one method which log the current catalog locale using the UserContext module.
+Now that we have our business logic implemented, we need to tell the system that our module has a dependency to ``pim/user-context`` and that we are exposing this object.
+
+The real thing
+++++++++++++++
+
+Here is how to do it using RequireJS:
+
+.. code-block:: javascript
+    :linenos:
+
+    // src/Acme/Bundle/CustomBundle/Resources/public/js/locale_logger.js
+    define(
+        ['pim/user-context'],
+        function (UserContext) {
+            return {
+                logLocale: function () {
+                    console.log(UserContext.get('uiLocale'));
+                }
+            }
+        }
+    )
+
+
+The ``define`` method takes two arguments: the first one is an array of the dependencies we want to use (``pim/user-context`` in this case) and a callback that will be called every time our module will be requested. This callback will receive as argument the list dependencies we asked for. Note that you can name the arguments of the callback as you want.
+
+Register it
++++++++++++
+
+Now that we have this small module, we would like to use it somewhere else. To do so, we need to register it in the RequireJS configuration file.
+
+.. code-block:: yaml
+
+    # src/Acme/Bundle/CustomBundle/Resources/config/requirejs.yml
+    config:
+        paths:
+            my_locale_logger: acmecustom/js/locale_logger
+
+
+We just declared a new module named ``my_locale_logger`` located in ``web/bundles/acmecustom/js/locale_logger.js``.
+Note that we don't need to specify the file extension and it is located in the public web folder (moved during the ``bin/console assets:install`` command execution). Also, ``my_locale_logger`` name is a arbitrary name and you can replace it with whatever you want.
+
+Make it available
++++++++++++++++++
+
+To make your module available you need to clear your cache:
+
+.. code-block:: bash
+
+    rm -rf ./var/cache/*
+
+
+dump the assets in the web folder
+
+.. code-block:: bash
+
+    bin/console assets:install web
+
+
+You can also decide to symlink those assets instead of copying them:
+
+.. code-block:: bash
+
+    bin/console assets:install --symlink web
+
+
+After that the last step is to build the bundle.js file
+
+.. code-block:: bash
+
+    npm run webpack
+
+
+This command will compile and minify all the pim files into web files and dump them in the public folder.
+
+You can also use the watch command which will recompile this file each time you modify a registered module.
+
+.. code-block:: bash
+
+    npm run webpack-watch
+
+
+Summary
++++++++
+
+We just created our first module and made it available for other part of the application. You can now use it by requirering ``my_locale_logger``.
```

Comment:
It miss "how to use it"... In fact, yes, we created a module, but we didn't see it in action.

Reply:
Yeah I agree about that, but as we cannot use it until line 219, we are kind of stuck. I will add an explanation about that :)